File: exercises/day13-Keyboard-First-Refactoring/java/src/test/java/sleightrip/trip/SleighTripServiceShould.java
package sleightrip.trip;

import sleightrip.exception.ElfNotLoggedInException;
import sleightrip.elf.Elf;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThatThrownBy;

class SleighTripServiceShould {

    @Test
    void throw_exception_when_no_elf_is_logged_in() {
        // given
        SleighTripService tripService = new SleighTripService() {
            protected Elf getLoggedUser() {
                return null; // no elf logged in
            }
        };
        Elf targetElf = new Elf();

        // when / then
        assertThatThrownBy(() -> tripService.getTripsByUser(targetElf))
                .isInstanceOf(ElfNotLoggedInException.class);
    }

    //TODO: Please finish testing the getTripsByUser method!!
}

File: exercises/day13-Keyboard-First-Refactoring/kotlin/src/test/kotlin/sleightrip/trip/SleighTripServiceShould.kt
package sleightrip.trip

import sleightrip.exception.ElfNotLoggedInException
import sleightrip.elf.Elf
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

class SleighTripServiceShould {

    @Test
    fun `throw exception when no elf is logged in`() {
        // given
        val tripService = object : SleighTripService() {
            override fun getLoggedUser(): Elf? {
                return null // no elf logged in
            }
        }
        val targetElf = Elf()

        // when / then
        assertThrows<ElfNotLoggedInException> {
            tripService.getTripsByUser(targetElf)
        }
    }

    //TODO: Please finish testing the getTripsByUser method!!
}
